### PR TITLE
fix freqkernel

### PR DESCRIPTION
**Pull request by JeffFessler**
Addresses #206 
The substance here is the changed lines with `wrapindex` and the corresponding added test.
There is also a subtle point that the kernel needs to be supported in [-N/2, N/2-1) so I added checks for that too (handling both even and odd N cases).  And some minor tweaking of docstrings.

FWIW, I'm teaching a graduate image processing course where I am trying to translate everything from Matlab to Julia so getting this `psf2tof` equivalent working is very desirable.

### CODE REVIEW

**Comment by JeffFessler** [comment]
Hi @johnnychen94 are you a maintainer here?  This PR has been ready for a while.
Anything else I should do?

File: src/utils.jl [hunk 1]
```diff
@@ -21,20 +20,36 @@ end
...
 The inverse of `freqkernel` is [`spacekernel`](@ref).
 """
 function freqkernel(::Type{T}, kern::AbstractArray, sz::Dims=size(kern); rfft=false) where T<:Union{Float32,Float64}
-    wrapindex(i, s) = i<1 ? i+s : i
-    all(size(kern) .<= sz) || throw(DimensionMismatch("kernel size $(size(kern)) is bigger than supplied size $sz"))
+    wrapindex(i, s) = 1 + (i<0 ? i+s : i)
+    all(size(kern) .<= sz) ||
+        throw(DimensionMismatch("kernel size $(size(kern)) exceeds supplied size $sz"))
+    rhs = Tuple(last(CartesianIndices(kern)))
+    rhs_limit = Int.(floor.((sz .+ 1) ./ 2)) # handle odd and even sizes
```
Comment (timholy) [inline src/utils.jl:46]:
```suggestion
    rhs_limit = (sz .+ 1) .÷ 2 # handle odd and even sizes
```

File: test/basic.jl [hunk 2]
```diff
@@ -109,6 +109,9 @@ end
     k2 = spacekernel(kfft, axes(k); rfftsz=31)
     @test k2 ≈ k
 
+    k = centered([0, 1, 0] * [0, 1, 0]') # Kronecker impulse
+    @test freqkernel(k, (8, 8)) ≈ ones(8,8) # flat frequency response
```
Comment (timholy) [inline test/basic.jl:113]: Would be good to test your exciting new size-checking code and make sure it returns the correct error messages.

**Review by timholy — COMMENTED** [review]
Sorry for the delay. It looks quite good, just a stylistic point (optional) and suggestion for testing your error conditions too.

File: src/utils.jl (hunk 1 above)
Comment (JeffFessler) [reply src/utils.jl:46]: oh cool!  i didn't know about ÷

File: src/utils.jl [hunk 3]
```diff
@@ -43,7 +43,7 @@ function freqkernel(::Type{T}, kern::AbstractArray, sz::Dims=size(kern); rfft=fa
     all(size(kern) .<= sz) ||
         throw(DimensionMismatch("kernel size $(size(kern)) exceeds supplied size $sz"))
     rhs = Tuple(last(CartesianIndices(kern)))
-    rhs_limit = Int.(floor.((sz .+ 1) ./ 2)) # handle odd and even sizes
+    rhs_limit = (sz .+ 1) .÷ 2 # handle odd and even sizes
     lhs = Tuple(first(CartesianIndices(kern)))
     lhs_limit = (-1) .* Int.(floor.((sz .+ 1) ./ 2))
```
Comment (timholy) [inline src/utils.jl:48]: one here too

Comment (JeffFessler) [reply src/utils.jl]: yep, and that line is kind of redundant so i've streamlined.

File: test/basic.jl (hunk 2 above)
Comment (JeffFessler) [reply test/basic.jl:113]: done

**Comment by JeffFessler** [comment]
I just learned about Aqua.jl and tested it out on ImageFiltering.jl and it reported that ImageMetadata is unused so I removed it from deps.  This is a bit off topic from freqkernel but figured I'd include it in this PR anyway and see how it goes...

**Comment by JeffFessler** [comment]
Also addressing #208 with a 1-line fix.  Hopefully.

**Comment by timholy** [comment]
Very nice work, @JeffFessler. Thanks!

**Comment by timholy** [comment]
https://github.com/JuliaRegistries/General/pull/30822

**Comment by johnnychen94** [comment]
I was AFK for the last two months so sorry for the inactivity. I've looked at the psf2otf source codes so I'm contaminated and thus can't review this PR (#81). Thank you both for the efforts.

**Pull request merged**